src/app/core/services/local-storage.service.ts

```diff
@@ -0,0 +1,23 @@
+import {Injectable} from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LocalStorageService<T> {
```

Comment:
remove the injectable. these two things don't mix: either it's an injectable or it is templated and instantiatable on demand